core/src/host.rs

```diff
@@ -62,29 +64,25 @@ pub trait EntryPointsCallerProvider {
 /// on a virtual machine or on a real blockchain.
 ///
 /// The `Deployer` trait provides a simple way to deploy a contract.
-pub trait Deployer: Sized {
+#[cfg(not(target_arch = "wasm32"))]
+pub trait Deployer<R: OdraContract>: Sized {
     /// Deploys a contract with given init args.
     ///
     /// If the init args are provided, the contract is deployed and initialized
     /// by calling the constructor. If the init args are not provided, the contract
     /// is deployed without initialization.
     ///
     /// Returns a host reference to the deployed contract.
-    fn deploy<T: InitArgs>(env: &HostEnv, init_args: T) -> Self;
+    fn deploy(env: &HostEnv, init_args: R::InitArgs) -> R::HostRef;
```

Comment:
It could but there is (and very likely it'll always be) a single implementation of `Deployer` - for each module - `T: OdraContract`. It could be a default implementation of `deploy` and `try_depoly` and empty `impl<R: OdraContract> Deployer<R> for R {}`.